internal/reminder/reminder.go

```diff
@@ -74,21 +78,52 @@ func (r *reminder) Start(ctx context.Context) error {
 		return errors.New("reminder stopped, cannot start again")
 	default:
 	}
+	defer r.Stop()
 
 	interval := r.cfg.RecurrenceConfig.Interval
 	if interval <= 0 {
 		return fmt.Errorf("invalid interval: %s", r.cfg.RecurrenceConfig.Interval)
 	}
 
+	metricsServerDone := make(chan struct{})
+
+	if r.cfg.MetricsConfig.Enabled {
+		metricsProviderReady := make(chan struct{})
+
+		go func() {
+			if err := r.startMetricServer(ctx, metricsProviderReady); err != nil {
+				logger.Fatal().Err(err).Msg("failed to start metrics server")
+			}
```

Comment:
This will call `exit()` even when `server.Shutdown(shutdownCtx)` return an error, is this correct? (This will prevent the deferred stop call)

Reply:
Since `startMetricServer` blocks, this seems reasonable.  The other option would be to have `startMetricServer` return a `chan error` and execute asynchronously within the function (i.e. move 68-86 into the goroutine started by `startMetricServer`).

Reply:
I feel the current option is fine.

I replaced `logger.Fatal().Err(err)` to be not `Fatal()`. OS will eventually clean up the socket and release the port.